woodwork/type_sys/inference_functions.py

```diff
@@ -45,7 +51,13 @@ def double_func(series):
 
 
 def boolean_func(series):
-    if pdtypes.is_bool_dtype(series.dtype):
+    if boolean_nullable_func(series) and series.notnull().all():
```

Comment:
Can we verify that this fast? I think it would be faster to say that if there is Any NaN value in the Boolean Column, then we use **BooleanNullable**. Otherwise use **Boolean** (Rather than checking all values are NaN)

Maybe we check performance with something like this: 
- https://stackoverflow.com/a/29530601/2512385

Reply:
How about `if boolean_nullable_func(series) and not series.isnull().any()` here instead?

Reply:
Yah that works